File: front-end/src/app/model/cart-response.ts
import { Product } from "./product";
import { UserCartResponse } from "./user-cart-response";

export class CartResponse {
  id:number;
  user:UserCartResponse;
  products:Product[];
  amount:number;
  quantity:number;

  constructor(id:number,user:UserCartResponse,
    products:Product[],amount:number,quantity:number){
      this.id=id;
      this.user=user;
      this.products=products;
      this.amount=amount;
      this.quantity=quantity;
    }
}

File: back-end/powerfit/src/main/java/com/nocountry/powerfit/model/response/CartResponse.java
package com.nocountry.powerfit.model.response;

import lombok.*;

import java.util.List;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class CartResponse {
    private Long id;
    private UserCartResponse user;
    private List<ProductResponse> products;
    private Double amount;
    private Integer quantity;
}
